File: 1877.cpp
#include <algorithm>

struct Solution {
    int minPairSum(vector<int>& nums) {
        int n = nums.size();
        sort(nums.begin(), nums.end());
        int res = nums[0] + nums[n - 1];
        for (int i = 1; i < n / 2 ; ++i){
            res = max(res, nums[i] + nums[n - 1 - i]);
        }
        return res;
    }
};

File: 1877.py
class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        res = nums[0] + nums[-1]
        for i in range(1, len(nums) // 2):
            res = max(res, nums[i] + nums[-1 - i])
        return res
